pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -179,45 +179,19 @@ func translateVirtualHost(
 // can be found. Called by translateRule to determine if
 func ConvertDestinationToCluster(destination *networking.Destination, vsvcName string,
 	in *networking.HTTPRoute, serviceIndex map[model.Hostname]*model.Service, defaultPort int) string {
-	// detect if it is a service
-	svc := serviceIndex[model.Hostname(destination.Host)]
-
-	if svc == nil {
-		if defaultPort != 80 {
-			noClusterMissingService.With(prometheus.Labels{
-				"service": destination.String(),
-				"rule":    vsvcName,
-			}).Add(1)
-			log.Infof("svc == nil => route on %d with missing cluster %s %v %v", defaultPort, vsvcName, destination, in)
-		}
-		return util.BlackHoleCluster
-	}
-
-	// default port uses port number
-	svcPort, _ := svc.Ports.GetByPort(defaultPort)
-
+	port := defaultPort
 	if destination.Port != nil {
 		switch selector := destination.Port.Port.(type) {
+		// TODO: remove port name from route.Destination in the API
 		case *networking.PortSelector_Name:
-			svcPort, _ = svc.Ports.Get(selector.Name)
+			log.Debuga("name based destination ports are not allowed => blackhole cluster")
```

Comment:
Should we log this as error so this useful msg is surfaced to the user?  User will be getting unhealthy upstream error when blackhole cluster is configured.

Reply:
This will be gone soon. Once this PR is in, we will remove the ability to specify destination ports by name

Reply:
SGTM, thanks!